src/virtual-source-buffer.js

```diff
@@ -718,16 +769,21 @@ export default class VirtualSourceBuffer extends videojs.EventTarget {
       });
 
       try {
-        destinationBuffer.updating = true;
         destinationBuffer.appendBuffer(tempBuffer);
       } catch (error) {
-        if (this.mediaSource_.player_) {
-          this.mediaSource_.player_.error({
-            code: -3,
-            type: 'APPEND_BUFFER_ERR',
-            message: error.message,
-            originalError: error
+        if (error instanceof DOMException && error.code === 22) {
+          let currGOPStart = currentGOPStart(this.gopBuffer_,
+                                             this.mediaSource_.player_.currentTime(),
+                                             this.timeMapping_);
+
+          this.trigger({
+            safeRemovePoint: currGOPStart,
+            segment: tempBuffer,
+            target: destinationBuffer,
+            type: 'bufferMaxed'
           });
+        } else {
```

Comment:
Sorry to be pendantic about this but we probably want to restore the old behavior in the else:

```
 if (this.mediaSource_.player_) {
  this.mediaSource_.player_.error({
  code: -3,
  type: 'APPEND_BUFFER_ERR',
  message: error.message,
  originalError: error
});
}
```

Also, it seems like `throw` is causing the Qunit tests to fail 😂

Reply:
Oh, forgot to mention the reason we want to do this is because it sets the error state on the tech, http://docs.videojs.com/tech#error, I believe this is also what shows an error dialog on the player

Reply:
Is that really appropriate, though? This won't be an `APPEND_BUFFER_ERR`; that would be the `QuotaExceededError` that we're catching above.

Not that I disagree with trying to catch uncaught errors for the purpose of notifying the user, generally speaking. But, there could be all sorts of errors thrown at various points in the code. Like the "source buffer already removed" error. So why do you think we should catch these in particular? And what sort of error should we re-throw to the player?

Reply:
Wouldn't it be an `APPEND_BUFFER_ERR` in the `else` clause though? My main reason for doing this is that we used to actually show all the errors you mention above("source buffer already removed" e.c.t). Otherwise those errors would be lost and the video element may end up in a bad playback with no way to tell that an error occurred

Reply:
eeeehhhhhhhhhhhh....... You're right that you used to show errors thrown by `appendBuffer`, but you can also have errors thrown by e.g. `remove`, or `addSourceBuffer`, which are just uncaught exceptions. They were before, they are now.

Is that bad? Yes.

Should they all be caught? Yes.

Should only some of them be caught in this PR, or all of them, or none of them?..... :man_shrugging:

Also, I don't think `APPEND_BUFFER_ERR` / code -3 is defined anywhere, even in videojs-errors.